pinax/ratings/tests/test_views.py

```diff
@@ -0,0 +1,98 @@
+from django.contrib.contenttypes.models import ContentType
+
+from .models import Car
+from .test import TestCase
+
+
+class RateViewTest(TestCase):
+    # tests for the RateView
+
+    def setUp(self):
+        self.test_user = self.make_user(username="test_user")
+        self.forester = Car.objects.create(name="Subaru Forester")
+
+    def test_post_rating_with_invalid_content_type_id(self):
+        # test line with code snippet:
+        #   get_object_or_404(ContentType, pk=self.kwargs.get("content_type_id"))
+        #
+        with self.login(self.test_user):
+            response = self.post(
+                "pinax_ratings:rate",
+                content_type_id=400,
+                object_id=1,
+                data={
+                    "rating": 3,
+                    "category": "handling"
+                },
+            )
+            self.response_404(response)
+
+    def test_post_rating_with_invalid_object_id(self):
```

Comment:
I understand your thinking writing this test... you're being careful and thorough. However, since `test_post_rating_with_invalid_content_type_id()` proved that bad "generic" data returns a 404, you really don't need to prove it again.

Why? The url kwargs `content_type_id` and `object_id` are always used as a pair, and if one is invalid and we get a 404 that's good enough. You can remove this test as the prior test did the job.

At the same time, rename the first test such that it's clear you're testing for response to an invalid generic object, not the actual url kwargs.

Reply:
@grahamu the test test_post_rating_with_invalid_content_type_id is removed and test_post_rating_with_invalid_content_type_id has been renamed